### PR TITLE
make event subscirptions work like event name prefix filter.

### PR DESCRIPTION
**Pull request by AntonPuko**

### CODE REVIEW

File: src/lib/EventPublisher.js [hunk 1]
```diff
@@ -44,6 +44,18 @@ type Subscription = {
 class EventPublisher extends EventEmitter {
   _subscriptionsByID: Map<string, Subscription> = new Map();
 
+  _emitWithPrefix = (eventName: string, event: Event): void => {
```
Comment (AntonPuko) [inline src/lib/EventPublisher.js]: not sure about the name

File: src/lib/EventPublisher.js [hunk 2]
```diff
@@ -44,6 +44,18 @@ type Subscription = {
...
+    this.eventNames()
+      .filter(
+        (eventNamePrefix: string): boolean =>
+          eventName.startsWith(eventNamePrefix),
+      )
+      .forEach(
+        (eventNamePrefix: string): void =>
```
Comment (jlkalberer) [inline src/lib/EventPublisher.js]: This should just be `eventName`

File: src/lib/EventPublisher.js [hunk 3]
```diff
@@ -77,13 +89,12 @@ class EventPublisher extends EventEmitter {
...
 
-    this.emit(eventData.name, event);
+    this._emitWithPrefix(eventData.name, event);
     this.emit(ALL_EVENTS, event);
   };
 
-
   subscribe = (
-    eventName: string = ALL_EVENTS,
+    eventNamePrefix: string = ALL_EVENTS,
```
Comment (jlkalberer) [inline src/lib/EventPublisher.js:65]: `= ALL_EVENTS` is never used

File: src/lib/EventPublisher.js (hunk 1 above)
Comment (jlkalberer) [reply src/lib/EventPublisher.js]: Name is fine but move the private functions after the public ones

**Review by jlkalberer — CHANGES_REQUESTED** [review]

**Pull request merged**